File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> indices;

        for(int i = 0; i < nums.size(); i++){
            indices[nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++){
            if(indices.count(nums[i]) && indices[nums[i]] != i){
                return true;
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        curr_sum = 0
        seen = {0: 1}
        for num in nums:
            curr_sum += num
            count += seen.get(curr_sum - k, 0)
            seen[curr_sum] = seen.get(curr_sum, 0) + 1
        return count